src/shared/context/TableContext/TableContext.stories.tsx

```diff
@@ -0,0 +1,1129 @@
+    const newUsers = [
+      {
+        id: `user-${Date.now()}`,
+        name: 'John Doe',
+        email: 'john.doe@company.com',
+        role: 'User',
+        status: 'Active',
+        lastLogin: '2024-01-17',
+      },
+    ]
+    dispatch({ type: 'SET_ROWS', payload: [...rows, ...newUsers] })
+  }
+
+  return (
+    <div className="bg-yellow-50 p-4 rounded-lg mb-6">
+      <h3 className="text-lg font-semibold mb-4">Story Controls</h3>
+
+      {/* Main Action Buttons */}
+      <div className="flex flex-wrap gap-2 mb-4">
+        <button
+          onClick={simulateLoading}
+          className="bg-blue-500 hover:bg-blue-700 text-white font-medium py-2 px-4 rounded text-sm"
+        >
+          Simulate Loading
+        </button>
+        <button
+          onClick={simulateError}
+          className="bg-red-500 hover:bg-red-700 text-white font-medium py-2 px-4 rounded text-sm"
+        >
+          Simulate Error
+        </button>
+        <button
+          onClick={resetTable}
+          className="bg-green-500 hover:bg-green-700 text-white font-medium py-2 px-4 rounded text-sm"
+        >
+          Reset Table
+        </button>
+        <button
+          onClick={addMoreData}
+          className="bg-purple-500 hover:bg-purple-700 text-white font-medium py-2 px-4 rounded text-sm"
+        >
+          Add More Data
+        </button>
+      </div>
+
+      {/* Quick Sort Demo */}
+      <div className="mb-4">
+        <h4 className="text-sm font-medium text-gray-700 mb-2">Quick Sort Demo</h4>
+        <div className="flex flex-wrap gap-2">
+          <button
+            onClick={() => dispatch({ type: 'SORT_BY_COLUMN', payload: { id: 'name', direction: 'asc' } })}
+            className="bg-indigo-500 hover:bg-indigo-700 text-white font-medium py-1 px-3 rounded text-xs"
+          >
+            Sort by Name ↑
+          </button>
+          <button
+            onClick={() => dispatch({ type: 'SORT_BY_COLUMN', payload: { id: 'email', direction: 'desc' } })}
+            className="bg-indigo-500 hover:bg-indigo-700 text-white font-medium py-1 px-3 rounded text-xs"
+          >
+            Sort by Email ↓
+          </button>
+          <button
+            onClick={() => dispatch({ type: 'SORT_BY_COLUMN', payload: { id: 'status', direction: 'asc' } })}
+            className="bg-indigo-500 hover:bg-indigo-700 text-white font-medium py-1 px-3 rounded text-xs"
+          >
+            Sort by Status ↑
+          </button>
+          <button
+            onClick={() =>
+              dispatch({ type: 'SORT_BY_COLUMN', payload: { id: 'lastLogin', direction: 'desc' } })
+            }
+            className="bg-indigo-500 hover:bg-indigo-700 text-white font-medium py-1 px-3 rounded text-xs"
+          >
+            Sort by Last Login ↓
+          </button>
+        </div>
+      </div>
+
+      {/* Custom Error Input */}
+      <div className="flex items-center space-x-2">
+        <input
```

Comment:
Im not sure if its my theme but the text inside the inputs are white and cannot be seen.